gnocchi/tests/test_indexer.py

```diff
@@ -984,7 +984,10 @@ def test_list_resources_custom_with_history(self):
         resources = sorted(
             [r.jsonify() for r in resources],
             key=operator.itemgetter("revision_start"))
-        self.assertEqual([r1, r2], resources)
+        expected_resources = sorted(
+            [r1, r2],
```

Comment:
err.. how is r1 have a later revision_start than r2? r1 is the create  and r2 is the update on the same resource?lol unless i'm reading this wrong?

Reply:
Agreed, I have blindly sorted the list, my sorted() doesn't sort anything. The culprit is somewhere else.

How does revision_end can be < revision_start...

b"  'revision_end': datetime.datetime(2017, 11, 9, 14, 50, 35, 96172, tzinfo=datetime.timezone.utc),"
b"  'revision_start': datetime.datetime(2017, 11, 9, 14, 50, 35, 297899, tzinfo=datetime.timezone.utc),"

Reply:
The code is here:
https://github.com/gnocchixyz/gnocchi/blob/master/gnocchi/indexer/sqlalchemy.py#L808-L814

On update we copy resource to resource history table and just update revision_end.

So that means the clock have gone backward between create and update. I didn't see another explanation.

Reply:
hm... does that mean r2's revision_start should always have a later timestamp than r1?